src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -186,6 +186,24 @@ private BoundExpression VisitExpressionImpl(BoundExpression node)
                     visited.Type.Equals(node.Type, TypeCompareKind.IgnoreDynamicAndTupleNames) ||
                     IsUnusedDeconstruction(node));
 
+            if (visited != null & visited != node)
```

Comment:
Did you mean `&&`?

Reply:
Short circuiting is not needed here so having no branches could make it a bit faster. 
But probably does not make a lot of difference though. I will change it to `&&`  

---
In reply to: [170113468](https://github.com/dotnet/roslyn/pull/24957#discussion_r170113468) [](ancestors = 170113468)